### PR TITLE
Added axios response parser in the backend utils

### LINKED ISSUE #656: Missing helper functions for processing axios response and error
`res.data.error.error_details.cause.cause.message` - thats a serious, and repetitive, drilldown.. perhaps some helper functions are in order?

_Originally posted by @shlomi-nx in https://github.com/nutanix/papiea/pull/636#discussion_r590832891_

### PR DESCRIPTION
**Pull request by nitesh-idnani1**
Closes #656
- Added helper modules in the axios response parser class.     
- Modified sdk and engine tests to use the axios response parser.

### CODE REVIEW

File: papiea-backend-utils/src/axios_response_parser.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        if (response) {
+            if (response.data) {
+                return response.status
+            }
+            console.info("Received null/undefined value for response status")
+        }
+        return 600
+    }
+    
+    public static getAxiosError(err: any): any {
```
Comment (Emulebest) [inline papiea-backend-utils/src/axios_response_parser.ts:36]: The return type should explicitly include `null`

File: papiea-backend-utils/src/axios_response_parser.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    
+    public static getAxiosErrorCode(err: any): number {
+        const axios_error = AxiosResponseParser.getAxiosError(err)
+        if (axios_error) {
+            if (axios_error.code) {
+                return axios_error.code
+            }
+            console.info("Received null/undefined value for error code")
+        }
+        return 600
```
Comment (Emulebest) [inline papiea-backend-utils/src/axios_response_parser.ts:55]: Why is this `600`?

File: papiea-backend-utils/src/axios_response_parser.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    
+    public static getAxiosErrorEntityInfo(err: any): any {
```
Comment (Emulebest) [inline papiea-backend-utils/src/axios_response_parser.ts:58]: Same, if `null` is an option for describing absence of value, it should be included in the return type signature

File: papiea-sdk/typescript/src/provider_sdk/typescript_sdk_utils.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import {Entity, ErrorSchemas, Provider} from "papiea-core"
 import {Request} from "express"
 import {FORMAT_HTTP_HEADERS, Span, Tracer} from "opentracing"
 import {ProceduralCtx_Interface} from "./typescript_sdk_interface"
+import e = require("express")
```
Comment (Emulebest) [inline papiea-sdk/typescript/src/provider_sdk/typescript_sdk_utils.ts:5]: We don't use `require` in the modern ts code. Please use `import * as express from "express"` instead

**Review by Emulebest — CHANGES_REQUESTED** [review]
LGTM, a few comments

File: papiea-sdk/typescript/src/provider_sdk/typescript_sdk_utils.ts (hunk 4 above)
Comment (nitesh-idnani1) [reply papiea-sdk/typescript/src/provider_sdk/typescript_sdk_utils.ts:5]: Yes, this line was added by mistake. Fixed it

File: papiea-backend-utils/src/axios_response_parser.ts (hunk 2 above)
Comment (nitesh-idnani1) [reply papiea-backend-utils/src/axios_response_parser.ts:55]: Need some default error code in case the parser failed.

**Pull request merged**